src/BizHawk.Emulation.Cores/Computers/Amiga/PUAE.cs

```diff
@@ -0,0 +1,261 @@
+﻿using BizHawk.Common;
+using BizHawk.Common.CollectionExtensions;
+using BizHawk.Common.StringExtensions;
+using BizHawk.Emulation.Common;
+using BizHawk.Emulation.Cores.Waterbox;
+
+using System;
+using System.Collections.Generic;
+
+namespace BizHawk.Emulation.Cores.Computers.Amiga
+{
+	[PortedCore(
+		name: CoreNames.PUAE,
+		author: "UAE Team",
+		portedVersion: "5.0.0",
+		portedUrl: "https://github.com/libretro/libretro-uae")]
+	public partial class PUAE : WaterboxCore
+	{
+		private LibPUAE _puae;
+		private static string _chipsetCompatible = "";
+
+		[CoreConstructor(VSystemID.Raw.Amiga)]
+		public PUAE(CoreLoadParameters<object, PUAESyncSettings> lp)
+			: base(lp.Comm, new Configuration
+			{
+				DefaultWidth          = LibPUAE.PAL_WIDTH,
+				DefaultHeight         = LibPUAE.PAL_HEIGHT,
+				MaxWidth              = LibPUAE.PAL_WIDTH,
+				MaxHeight             = LibPUAE.PAL_HEIGHT,
+				MaxSamples            = 2 * 1024,
+				SystemId              = VSystemID.Raw.Amiga,
+				DefaultFpsNumerator   = 50,
+				DefaultFpsDenominator = 1
+			})
+		{
+			_syncSettings = lp.SyncSettings ?? new();
+
+			_puae = PreInit<LibPUAE>(new WaterboxOptions
+			{
+				Filename                   = "puae.wbx",
+				SbrkHeapSizeKB             = 5 * 512,
+				SealedHeapSizeKB           = 10 * 1024,
+				InvisibleHeapSizeKB        = 10 * 1024,
+				PlainHeapSizeKB            = 10 * 1024,
+				MmapHeapSizeKB             = 40 * 1024,
+				SkipCoreConsistencyCheck   = lp.Comm.CorePreferences.HasFlag(CoreComm.CorePreferencesFlags.WaterboxCoreConsistencyCheck),
+				SkipMemoryConsistencyCheck = lp.Comm.CorePreferences.HasFlag(CoreComm.CorePreferencesFlags.WaterboxMemoryConsistencyCheck),
+			});
+
+			var filesToRemove = new List<string>();
+			var args = CreateArguments(_syncSettings);
+
+			_exe.AddReadonlyFile(lp.Roms[0].FileData, "romfile");
+			filesToRemove.Add("romfile");
+
+			var (kickstartData, kickstartInfo) = CoreComm.CoreFileProvider.GetFirmwareWithGameInfoOrThrow(
+				new(VSystemID.Raw.Amiga, _chipsetCompatible),
+				"Firmware files are usually required and may stop your game from loading");
+			_exe.AddReadonlyFile(kickstartData, kickstartInfo.Name);
+			filesToRemove.Add(kickstartInfo.Name);
+			args.AddRange(new List<string>
+			{
+				"-r", kickstartInfo.Name
+			});
+
+			ControllerDefinition = InitInput();
+
+			if (!_puae.Init(args.Count, args.ToArray()))
+				throw new InvalidOperationException("Core rejected the rom!");
+
+			foreach (var s in filesToRemove)
+			{
+				//_exe.RemoveReadonlyFile(s);

```

Comment:
Is this supposed to be commented out?

Reply:
The core seems to need some of those files during emulation, I wanted to figure out which ones I can release later.